enclave-runner/src/usercalls/mod.rs

```diff
@@ -333,36 +348,258 @@ impl EnclaveState {
             last_fd,
             exiting: AtomicBool::new(false),
             usercall_ext : usercall_ext,
+            work_sender: Mutex::new(work_sender)
         })
     }
 
+    fn create_worker_threads(num_of_threads: usize,
+                             work_receiver: crossbeam::crossbeam_channel::Receiver<Work>,
+                             io_queue_send: mpsc::Sender<UsercallSendData>,
+                             enclave: Arc<EnclaveState>) {
+        for _ in 1..num_of_threads {
```

Comment:
This will just result in confusing off-by-one errors. This function should just take in the number of worker threads. Then you can also get rid of the minimum `num_of_threads` in the other function.